feat: add C ndarray interface and refactor implementation for `stats/base/snanmskrange` (#7238)
stdlib-js/stdlib: lib/node_modules/@stdlib/stats/base/snanmskrange/README.md

-   **N**: number of indexed elements.
-   **x**: input [`Float32Array`][@stdlib/array/float32].
-   **strideX**: stride length for `x`.
-   **mask**: mask [`Uint8Array`][@stdlib/array/uint8]. If a `mask` array element is `0`, the corresponding element in `x` is considered valid and **included** in computation. If a `mask` array element is `1`, the corresponding element in `x` is considered invalid/missing and **excluded** from computation.
-   **strideMask**: stride length for `mask`.

> The `N` and stride parameters determine which elements in the strided array are accessed at runtime. For example, to compute the [range][range] of every other element in `x`,

Reviewer: ```suggestion
The `N` and stride parameters determine which elements in the strided arrays are accessed at runtime. For example, to compute the [range][range] of every other element in `x`,
```

More than one strided array.

-   **offsetX**: starting index for `x`.
-   **offsetMask**: starting index for `mask`.

> While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, the offset parameter supports indexing semantics based on a starting index. For example, to calculate the [range][range] for every other element in `x` starting from the second element

Reviewer: ```suggestion
While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, offset parameters support indexing semantics based on starting indices. For example, to calculate the [range][range] for every other element in `x` starting from the second element
```

```c
#include <stdint.h>

const float x[] = { 1.0f, -2.0f, 4.0f, 2.0f, 0.0f/0.0f };
const uint8_t mask[] = { 0, 0, 1, 0, 0 };

float v = stdlib_strided_snanmskrange( 5, x, 1, mask, 1 );
// returns 4.0f
```

The function accepts the following arguments:

-   **N**: `[in] CBLAS_INT` number of indexed elements.
-   **X**: `[in] float*` input array.
-   **strideX**: `[in] CBLAS_INT` stride length for `X`.
-   **Mask**: `[in] uint8_t*` mask array. If a `Mask` array element is `0`, the corresponding element in `X` is considered valid and included in computation. If a `Mask` array element is `1`, the corresponding element in `X` is considered invalid/missing and excluded from computation.
-   **strideMask**: `[in] CBLAS_INT` stride length for `Mask`.

```c
float stdlib_strided_snanmskrange( const CBLAS_INT N, const float *X, const CBLAS_INT strideX, const uint8_t *Mask, const CBLAS_INT strideMask );
```

#### stdlib_strided_snanmskrange_ndarray( N, \*X, strideX, offsetX, \*Mask, strideMask, offsetMask )

Computes the [range][range] of a single-precision floating-point strided array according to a `mask`, ignoring `NaN` values and using alternative indexing semantics.

```c

> const float x[] = { 1.0f, -2.0f, 4.0f, 2.0f, 0.0f/0.0f };

Reviewer: ```suggestion
#include <stdint.h>

const float x[] = { 1.0f, -2.0f, 4.0f, 2.0f, 0.0f/0.0f };
```
